src/components/profile-content/profile-content.js

```diff
@@ -79,33 +100,81 @@ const ProfileContent = (props) => {
         props.setBookmarkedRulesCount(bookmarkedRulesSpread.length);
       })
       .catch((err) => {
-        console.error('error: ', err);
+        appInsights.trackException({
+          error: new Error(err),
+          severityLevel: 3,
+        });
       });
   }
 
-  function getUserComments() {
-    GetUserComments(user.nickname, commentsRepository)
-      .then((success) => {
-        const allRules = props.data.allMarkdownRemark.nodes;
-        const commentGuids =
-          success.items.size != 0 ? success.items.map((r) => r.title) : null;
-        const commentedRulesMap = allRules.filter((value) =>
-          commentGuids.includes(value.frontmatter.guid)
-        );
-        const commentedRulesSpread = commentedRulesMap.map((r) => ({
-          ...r.frontmatter,
-          excerpt: r.excerpt,
-          htmlAst: r.htmlAst,
-          url: success.items
-            .filter((v) => v.title == r.frontmatter.guid)
-            .map((r) => r.html_url)[0],
-        }));
-        setCommentedRulesList(commentedRulesSpread);
-        props.setCommentedRulesCount(commentedRulesSpread.length);
-      })
-      .catch((err) => {
-        console.error('error: ', err);
-      });
+  async function getGuidFromDisqus(comment) {
+    if (comment.forum == process.env.DISQUS_FORUM) {
+      const guid = await GetCommentSlug(comment.thread)
+        .then((success) => {
+          return success.response.identifiers[0];
+        })
+        .catch((err) => {
+          appInsights.trackException({
+            error: new Error(err),
+            severityLevel: 3,
+          });
+        });
+
+      return guid;
+    }
+  }
+
+  async function createGuidList(response) {
+    var guids = response.map((comment) => {
+      return getGuidFromDisqus(comment);
+    });
+
+    return await Promise.all(guids);
+  }
+
+  async function setCommentedRulesFromGuids(reponse) {
+    const commentedRuleGuids = await createGuidList(reponse);
```

Comment:
This section is concerning because it is going to hit the disqus api for every comment the user has made. I think we can do it better by calling list posts for the user with the related thread argument. We can then use the identifiers on the related thread to identify the associated rule.

Reply:
Thanks for the tip, I made this change in this commit 5664366499c0510fee388e2036cd93d18d203dc9